<launch>

    <!-- this input client can operate using either absolute xy position setpoints or relative xy position setpoints (absolute by default) -->
    <!-- to use relative xy position setpoints, simply pass a true value to the argument "use_relative_xy_setpoints" -->
    <!-- when using relative setpoints, the reference (0,0) position is set to the position at which the vehicle entered offboard mode -->
    <!-- note that this reference (0,0) will be set every time the vehicle enters offboard mode (not just the first time) -->
    <!-- note that both the z position (altitude) and the yaw angle are always defined using absolute position for safety reasons -->

    <arg name="cmd_rate_hz" default="30"/>
    <arg name="use_relative_xy_setpoints" default="false"/>
    <arg name="sub_topic_fcu_state" default="/aqc/fcu_state"/>
    <arg name="pub_topic_position_setpoint" default="/aqc/cmd_pos"/>
    <arg name="sub_topic_quad_pos" default="/aqc/quad_state/position"/>
    <arg name="dr_config_file" default="$(find aqc_input_dynamic_reconfigure_pos)/config/dr_default.yaml"/> 

    <param name="/aqc_input_dynamic_reconfigure_pos/cmd_rate_hz" value="$(arg cmd_rate_hz)"/>
    <param name="/aqc_input_dynamic_reconfigure_pos/use_relative_xy_setpoints" value="$(arg use_relative_xy_setpoints)"/>

    <!-- launch dynamic reconfigure input client -->
    <node pkg="aqc_input_dynamic_reconfigure_pos" type="aqc_input_client_dr_pos_node" name="aqc_input_client_dr_pos_node">
        <remap from="/cmd_position" to="$(arg pub_topic_position_setpoint)"/>
        <remap from="/fcu_state" to="$(arg sub_topic_fcu_state)"/>
        <remap from="/quad_state/position" to="$(arg sub_topic_quad_pos)"/>
    </node>

    <!-- launch rqt dynamic reconfiguration gui and load default parameters from config file -->
    <node name="rqt_reconfigure_gui" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen"/>
    <node name="reconfigure_load_params" pkg="dynamic_reconfigure" type="dynparam" args="load /aqc_input_client_dr_pos_node $(arg dr_config_file)"/>

</launch>
